charts/zwavejs2mqtt/2.0.0/questions.yaml

```diff
@@ -0,0 +1,522 @@
+                  type: string
+                  default: "config"
+                  required: true
+                  editable: false
+                  hidden: true
+              - variable: mountPath
+                label: "Mount path inside the container"
+                schema:
+                  type: path
+                  default: "/usr/src/app/store"
+                  editable: false
+              - variable: hostPathEnabled
+                label: "Config - Enable Host Path"
+                description: "something about this here"
+                schema:
+                  type: boolean
+                  default: false
+                  show_subquestions_if: true
+                  subquestions:
+                    - variable: hostPath
+                      label: "Config Host Path"
+                      description: "Host path to store config"
+                      schema:
+                        type: hostpath
+                        required: true
+        # Z-Wave device -----------------
+        - variable: zwave
+          label: "USB Z-Wave Device"
+          schema:
+            type: dict
+            attrs:
+              - variable: enabled
+                label: "Enabled"
+                schema:
+                  type: boolean
+                  default: true
+                  required: true
+                  hidden: true
+                  editable: false
+              - variable: setPermissions
+                label: "Automatic Permissions"
+                description: "Automatically set permissions on install"
+                schema:
+                  type: boolean
+                  default: false
+                  hidden: true
+              - variable: name
+                label: "Mountpoint Name"
+                schema:
+                  type: string
+                  default: "ZWave"
+                  required: true
+                  editable: true
+              - variable: emptyDir
+                label: "emptyDir"
+                schema:
+                  type: boolean
+                  default: false
+                  hidden: true
+                  editable: false
+              - variable: mountPath
+                label: "Path to mount device inside the container"
+                description: "Preferably the same as the path to the device path on the host"
+                schema:
+                  type: path
+                  required: true
+                  default: "/dev/ttyUSB0"
+                  editable: true
+              - variable: hostPathEnabled
+                label: "host Path Enabled"
+                schema:
+                  type: boolean
+                  default: true
+                  hidden: true
+              - variable: hostPath
+                label: "Path to Z-Wave device on host"
+                schema:
+                  type: path
+                  required: true
+                  default: "/dev/ttyUSB0"
```

Comment:
#### Primary error this suggestion resolves

```
Error: [EFAULT] Failed to install chart release: Error: failed pre-install: warning: Hook pre-install zwavejs2mqtt/templates/common.yaml failed: Job.batch "ztest-zwavejs2mqtt-config" is invalid: [spec.template.spec.volumes[1].name: Invalid value: "ZWave": a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name', or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?'), spec.template.spec.containers[0].volumeMounts[1].name: Not found: "ZWave"]
```

- the offending variable

```
              - variable: name 
                label: "Mountpoint Name"
                schema:
                  type: string
                  default: "ZWave"
                  required: true
                  editable: true
```

I've just removed the entire `- variable: name` - The way I made `_appStorage.tpl` was to make including a name optional, so it would also work with a list  ( I'm sure you know this already - I'm saying it here to be corrected, in case I am wrong ) If this was a list, `range $name` would be an index integer so we have the option to correctly set a name with ` $name = $av.name` -- But with our apps, each mount is a dictionary, so we can just leave the name set to match the dictionary's name (provided for reference)
```
{{- range $name, $av := $.volMounts -}}
{{ if (default true $av.enabled) }}
{{ if $av.name }}
{{ $name = $av.name }}
{{ end }}
```

### The remaining changes are for improvement only

```
              - variable: hostPath
                label: "Path to Z-Wave device on host"
                schema:
                  type: hostpath
                  required: true
                  default: "/dev/ttyUSB0"
```

I changed `default: "/dev/ttyUSB0"` to `default: "/dev/ttyACM0` because that's how the Z-Wave sticks I have show up. IMO, this is closer to an "out-of-the-box" setting that will work. In any case, most users will still want to use `/dev/serial/by-id/...`

I also changed `type: hostpath` to `type: path` since the former is limited to the `/mnt` folder browser and we can not use it to pick a USB  device.

![image](https://user-images.githubusercontent.com/35702532/111222933-d9390e00-85b2-11eb-8f98-0e2d4ab4dd0a.png)

Remaining changes in my suggestion were just clean-up to remove lines not required, since most of the variables in this section are hidden anyways.

```suggestion
        # Z-Wave device -----------------
        - variable: zwave
          label: "USB Z-Wave Device"
          schema:
            type: dict
            attrs:
              - variable: hostPathEnabled
                label: "Enable Host Path"
                schema:
                  type: boolean
                  default: true
                  hidden: true
              - variable: setPermissions
                label: "Automatic Permissions"
                schema:
                  type: boolean
                  default: false
                  hidden: true
              - variable: emptyDir
                label: "Empty Dir"
                schema:
                  type: boolean
                  default: false
                  hidden: true
              - variable: mountPath
                label: "Path to mount Z-Wave device inside the container"
                description: "Preferably the same as the path to the device path on the host"
                schema:
                  type: path
                  default: "/dev/ttyACM0"
                  required: true
              - variable: hostPath
                label: "Path to Z-Wave device on host"
                schema:
                  type: path
                  default: "/dev/ttyACM0"
                  required: true
```